src/tech/v3/dataset/impl/column.clj

```diff
@@ -329,7 +329,9 @@
       :spearman (dfn/spearmans-correlation col other-column)
       :kendall (dfn/kendalls-correlation col other-column)))
   (select [col idx-rdr]
-    (let [idx-rdr (->efficient-reader idx-rdr)]
+    (let [idx-rdr (if (= (dtype/elemwise-datatype idx-rdr) :boolean)
+                    (->efficient-reader (keep-indexed #(when (true? %2) %1) idx-rdr))
```

Comment:
I think there may better way to do this - 
[bool-reader->indexes](https://github.com/cnuernber/dtype-next/blob/master/src/tech/v3/datatype/unary_pred.clj#L267).  You will have to ensure it is a reader before calling that method - [ensure-reader](https://cnuernber.github.io/dtype-next/tech.v3.datatype.argops.html#var-ensure-reader).

It is conceptually the same as [argfilter](https://cnuernber.github.io/dtype-next/tech.v3.datatype.argops.html#var-argfilter) - `(argops/argfilter identity idx-rdr).

Reply:
Oh nice. I was wondering if there was a more efficient way to do this.